zebra-state/src/service/finalized_state/disk_format/upgrade/cache_genesis_roots.rs

```diff
@@ -0,0 +1,181 @@
+//! Updating the genesis note commitment trees to cache their roots.
+//!
+//! This reduces CPU usage when the genesis tree roots are used for transaction validation.
+//! Since mempool transactions are cheap to create, this is a potential remote denial of service.
+
+use std::sync::mpsc;
+
+use zebra_chain::{block::Height, sprout};
+
+use crate::service::finalized_state::{disk_db::DiskWriteBatch, ZebraDb};
+
+use super::CancelFormatChange;
+
+/// Runs disk format upgrade for changing the sprout and history tree key types.
+///
+/// Returns `Ok` if the upgrade completed, and `Err` if it was cancelled.
+///
+/// # Panics
+///
+/// If the state is empty.
+#[allow(clippy::unwrap_in_result)]
+#[instrument(skip(upgrade_db, cancel_receiver))]
+pub fn run(
+    initial_tip_height: Height,
```

Comment:
###### Optional Suggestion

```suggestion
    _initial_tip_height: Height,
```